code/modules/clothing/glasses/glasses.dm

```diff
@@ -230,3 +232,27 @@
 	icon_state = "redglasses"
 	item_state = "redglasses"
 
+
+/obj/item/clothing/glasses/proc/chameleon(var/mob/user)
+	var/list/glasses_types = typesof(/obj/item/clothing/glasses) - src.type - /obj/item/clothing/glasses/hud - /obj/item/clothing/glasses/night/shadowling - /obj/item/clothing/glasses/hud/toggle - /obj/item/clothing/glasses
+	var/list/glasses = list()
+
+	// Generate them into a list
+	for(var/glasses_type in glasses_types)
+		var/obj/item/clothing/glasses/S = new glasses_type
+		glasses[capitalize(S.name)] = S
+
+	var/list/show_glasses = list("EXIT" = null) + sortList(glasses) // the list that will be shown to the user to pick from
+
+	var/input_glasses = input(user, "Choose a piece of eyewear to disguise as.", "Choose glasses style.") in show_glasses
+
+	if(user && src in user.contents)
+
+		var/obj/item/clothing/glasses/chosen_glasses = glasses[capitalize(input_glasses)]

```

Comment:
Says it's an invalid expression. I assume I'm doing something wrong but double checking with you to make sure you wrote it correctly.